include/cuco/probing_scheme.cuh

```diff
@@ -175,6 +177,159 @@ class double_hashing : private detail::probing_scheme_base<CGSize> {
+   *
+   * @tparam NewHash New hasher type
+   *
+   * @param hash Hasher
+   *
+   * @return Copy of the current probing method
+   */
+  template <typename NewHash>
+  [[nodiscard]] __host__ __device__ constexpr auto with_hash_function(
+    NewHash const& hash) const noexcept;
+
+  /**
+   * @brief Operator to return a probing iterator
+   *
+   * @tparam ProbeKey Type of probing key
+   * @tparam Extent Type of extent
+   *
+   * @param probe_key The probing key
+   * @param upper_bound Upper bound of the iteration
+   * @return An iterator whose value_type is convertible to slot index type
+   */
+  template <typename ProbeKey, typename Extent>
+  __host__ __device__ constexpr auto operator()(ProbeKey const& probe_key,
+                                                Extent upper_bound) const noexcept;
+
+  /**
+   * @brief Operator to return a CG-based probing iterator
+   *
+   * @tparam ProbeKey Type of probing key
+   * @tparam Extent Type of extent
+   *
+   * @param g the Cooperative Group to generate probing iterator
+   * @param probe_key The probing key
+   * @param upper_bound Upper bound of the iteration
+   * @return An iterator whose value_type is convertible to slot index type
+   */
+  template <typename ProbeKey, typename Extent>
+  __host__ __device__ constexpr auto operator()(
+    cooperative_groups::thread_block_tile<cg_size> const& g,
+    ProbeKey const& probe_key,
+    Extent upper_bound) const noexcept;
+
+  /**
+   * @brief Perfect probing iterator class.
+   *
+   * @tparam Extent Type of Extent
+   */
+  template <typename Extent>
+  class probing_iterator {
+   public:
+    using extent_type = Extent;                            ///< Extent type
+    using size_type   = typename extent_type::value_type;  ///< Size type
+
+    /**
+     * @brief Constructs an probing iterator
+     *
+     * @param start Iteration starting point
+     * @param upper_bound Upper bound of the iteration
+     */
+    __host__ __device__ constexpr probing_iterator(size_type start,
+                                                   extent_type upper_bound) noexcept
+      : curr_index_{start}, upper_bound_{upper_bound}
+    {
+    }
+
+    /**
+     * @brief Dereference operator
+     *
+     * @return Current slot index
+     */
+    __host__ __device__ constexpr auto operator*() const noexcept { return curr_index_; }
+
+    /**
+     * @brief Prefix increment operator
+     *
+     * @return Current iterator
+     */
+    __host__ __device__ constexpr auto operator++() noexcept
+    {
+      // TODO: This is still getting called, and should never really be used
+      curr_index_ = (curr_index_ + 1) % upper_bound_;
+      // curr_index_ = 0;
+      return *this;
+    }
+
+    /**
+     * @brief Postfix increment operator
+     *
+     * @return Old iterator before increment
+     */
+    __host__ __device__ constexpr auto operator++(int32_t) noexcept
+    {
+      auto temp = *this;
```

Comment:
also static_assert that this is not allowed.

Reply:
Sounds good!

Reply:
Actually, you can't `static_assert` in this function because that will just cause compilation to fail even if the function isn't called.  I think that you'll actually just be deleting the function.  If you want, you can put a comment about the fact that incrementing is not allowed.

If the fact that this function is missing causes a compilation error then it is the caller that is wrong.